scripts/hooks/deploy.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/sh
+pwd
+main() {
+    echo "Deploying..."
+
+    if [ -z "${HOOK_tag}" ]; then
+      echo "The tag is not set. Skipping deployment."
+      exit 1
+    fi
+
+    if [ -z "${HOOK_image}" ]; then
+      echo "The image is not set. Skipping deployment."
+      exit 1
+    fi
+
+    if [ -z "${HOOK_service}" ]; then
+      echo "The service is not set. Skipping deployment."
+      exit 1
+    fi
+
+    echo "Updating service ${HOOK_service}..."
+
+    cmd="docker service update --image ${HOOK_image}:${HOOK_tag} ${HOOK_service}"
```

Comment:
We better harden this so it's safe to run.  We can use something like `printf %q` to safely transform the values into something we can put in a command-line arg.  From the [printf man page](https://man7.org/linux/man-pages/man1/printf.1.html):

>        %q     ARGUMENT is printed in a format that can be reused as
>               shell input, escaping non-printable characters with the
>               proposed POSIX $'' syntax.

```sh
docker service update --image "$(printf '%q' "${HOOK_image}:${HOOK_tag}")" "$(printf '%q' "${HOOK_service}")"
```

Also, let's use `ALL_CAPS` for the environment variables.

Reply:
Oh this is neat, I didn't know about the `printf %q`.

The environment variables are automatically created from the hooks.json file, the variables were lowercase there, hence the lowercase envVars. Will change them to uppercase in hooks.json

Reply:
Brought this up in Slack, but `printf %q` is not POSIX compliant, therefore our Busybox alpine image will not be able to run scripts containing this.

https://www.shellcheck.net/wiki/SC3050